content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,72 +1,48 @@
-
+A socket is an **endpoint** of communication, acting as a connection between two computers on a network. It contains a socket number that the TCP layer can use to identify the application to which data will be transmitted. Additionally, an endpoint is made up of a port number and an IP address.
 There are two types of sockets
-
 1. Server socket - which waits for a request from clients
 2. Client socket - can be used to send and receive a request from clients.
-
 The client must know two information about the server:
 1.IP address of Server and 
 2.Port number
-
->Port number 0-1024 are reserved for administrative purposes(e,g 21 for FTP,23 for Telnet,25 for email and 80 for HTTP).In our program we will use port number 5000.
-
+>Port number 0-1024 are reserved for administrative purposes(e,g 21 for FTP,23 for Telnet,25 for email and 80 for HTTP).In our program, we will use port number 5000.
 Whenever we create a socket connection in Java, the following steps take place:
-
 1. Server creates a `ServerSocket` object, to indicate which port number our communication is to occur. If an I/O  error occurs, exception handling mechanisms are implemented; **Throws** an **I/OException** or a **try** **catch** block is used.
 2. The `accept()` function of the ServerSocket class is called by the server. This method waits for a client to connect to the server on a specific port before returning.
 3. A client builds a `Socket` object, giving the name of the server and the connecting port number as the server awaits.
 4. The `Socket` class constructor will try to connect the client to the server and port number provided. The client now has a `socket` object that if communication is established, may connect with the server.
 5. The `accept()` method on the server produces a reference to a new server socket that is connected to the client's socket.
  
-If the connection is successful, client and server can then communicate using **I/O streams**. The **I/O streams** come from socket objects of the client and server. The client's `OutputStream` connects to the server's `InputStream` while the server's `OutputStream` connects to the client's `InputStream`.
-
-Incase you don't know, a stream is a collection sequenced data. They can be divided into two categories:
+If the connection is successful, the client and server can then communicate using **I/O streams**. The **I/O streams** come from socket objects of the client and server. The client's `OutputStream` connects to the server's `InputStream` while the server's `OutputStream` connects to the client's `InputStream`.
+In case you don't know, a stream is a collection of sequenced data. They can be divided into two categories:
 1. Character stream (usually used with text files)
```

Comment:
Change:

> 1. Character stream (usually used with text files)
> 2. Byte stream (usually used with images) Character stream is in human-readable language while byte stream is in machine language, a language understandable by the CPU.
> In our case, the client and server will send messages to each other's streams in parallel. This is made possible because TCP is a two-way communication protocol.

To:

> 1. Character stream (usually used with text files)
> 2. Byte stream (usually used with images)
> 
> A character stream is in human-readable language while a byte stream is in machine language, a language understandable by the CPU. In our case, the client and server will send messages to each other's streams in parallel. This is made possible because TCP is a two-way communication protocol.

The paragraph after the description of a byte stream:

>Character stream is in human-readable language while byte stream is in machine language, a language understandable by the CPU. In our case, the client and server will send messages to each other's streams in parallel. This is made possible because TCP is a two-way communication protocol.

Should be on a new line.